src/reconcile/rec_write.c

```diff
@@ -2528,8 +2528,13 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
     if (last_block && r->multi_next == 1 && block_meta->page_id != WT_BLOCK_INVALID_PAGE_ID &&
       block_meta->delta_count < btree->max_consecutive_delta) {
         WT_RET(__rec_build_delta(session, r, chunk->image.mem, &build_delta));
-        /* Discard the delta if it is larger than one tenth of the size of the full image. */
-        if (build_delta && ((r->delta.size * 100) / chunk->image.size) > btree->delta_pct)
+        /*
+         * Discard the delta if it is larger than the maximum allowable delta size, or it is larger
+         * than one tenth of the size of the full image.
+         */
+        if (build_delta &&
+          (r->delta.size > btree->max_delta_size ||
```

Comment:
We should check the size of the full page not the size of delta.

Reply:
@quchenhao I understand what you want to implement, but I'm not understanding why we want to do this.

Reply:
If the full page is very small, writing a delta saves us very small amount of space but we pay for the cost to reconstruct it when reading.

Reply:
That makes sense. Is this what Ashish and Abishek wanted? I had a different understanding.

Reply:
yeah, we should clarify that with them first.